receivers/sns/config.go

```diff
@@ -0,0 +1,88 @@
+package sns
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/grafana/grafana-aws-sdk/pkg/awsds"
+
+	"github.com/grafana/alerting/receivers"
+	"github.com/grafana/alerting/templates"
+)
+
+type Config struct {
+	Topic           string
+	Subject         string
+	Body            string
+	MessageFormat   string
+	AWSAuthSettings awsds.AWSDatasourceSettings
+}
+
+func NewConfig(jsonData json.RawMessage, decryptFn receivers.DecryptFunc) (Config, error) {
+	type snsSettingsRaw struct {
+		Topic         string `json:"topic,omitempty" yaml:"topic,omitempty"`
+		AuthProvider  string `json:"authProvider,omitempty" yaml:"authProvider,omitempty"`
+		Subject       string `json:"subject,omitempty" yaml:"subject,omitempty"`
+		Body          string `json:"body,omitempty" yaml:"body,omitempty"`
+		MessageFormat string `json:"messageFormat,omitempty" yaml:"messageFormat,omitempty"`
+		Credentials   string `json:"credentials,omitempty" yaml:"credentials,omitempty"`
+		AccessKey     string `json:"accessKey,omitempty" yaml:"accessKey,omitempty"`
+		SecretKey     string `json:"secretKey,omitempty" yaml:"secretKey,omitempty"`
+		AssumeRoleARN string `json:"assumeRoleARN,omitempty" yaml:"assumeRoleARN,omitempty"`
+		ExternalID    string `json:"externalId,omitempty" yaml:"externalId,omitempty"`
```

Comment:
First, thank you so much for opening the PR! 

I have not reviewed it yet but the first thing that catches my eye is the config. We're trying to maintain as little drift as possible with integration in the upstream. Please make sure that json\yaml fields match the upstream integration 
https://github.com/prometheus/alertmanager/blob/f82574a3763c21ca9bf2ada7baf69bd09e01d977/config/notifiers.go#L731-L744

Reply:
Thanks for sharing, I will look at that. I think I will need to use this [SubformOptions](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/notifier/channels_config/plugin.go#L27) to provide the complex types like [sigv4.SigV4Config](https://github.com/prometheus/common/blob/main/sigv4/sigv4_config.go#L25)?

Reply:
Yes, complex types can be presented via subform options. You can see https://github.com/grafana/grafana/blob/cf601fab0906f1d811f1aa2fe18a67bc5421e3db/pkg/services/ngalert/notifier/channels_config/available_channels.go#L1312-L1338 as an example

Reply:
Ok I updated the config to match the upstream and pulled in some of the functionality from there too, like message deduplication and truncation.

Also I noticed that the Grafana channel config UIs doesn't seem to support ShowWhen or secure fields for subforms. I can try to address that in my PR to grafana.